packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -36,134 +35,139 @@ interface InMemoryAnchorConfig {
+    this.#anchorDelay = _config?.anchorDelay ?? 0;
+    this.#anchorOnRequest = _config?.anchorOnRequest ?? true;
+    this.#verifySignatures = _config?.verifySignatures ?? true;
   }
 
   async init(): Promise<void> {
-    return
+    return;
   }
 
   /**
    * @returns An array of the CAIP-2 chain IDs of the blockchains that are supported by this
    * anchor service
    */
   async getSupportedChains(): Promise<Array<string>> {
-    return [CHAIN_ID]
+    return [CHAIN_ID];
   }
 
   /**
    * Anchor requests
    */
   async anchor(): Promise<void> {
-    const candidates = await this._findCandidates()
+    const candidates = await this._findCandidates();
     for (const candidate of candidates) {
-      await this._process(candidate)
+      await this._process(candidate);
     }
 
-    this._queue = [] // reset
+    this.#queue = []; // reset
   }
 
-
-    /**
+  /**
    * Filter candidates by document and DIDs
    * @private
    */
   async _findCandidates(): Promise<Candidate[]> {
-    const groupedCandidates = await this._groupCandidatesByDocId(this._queue)
-    return await this._selectValidCandidates(groupedCandidates)
+    const groupedCandidates = await this._groupCandidatesByDocId(this.#queue);
+    return this._selectValidCandidates(groupedCandidates);
   }
 
   async _groupCandidatesByDocId(candidates: Candidate[]): Promise<Record<string, Candidate[]>> {
     const result: Record<string, Candidate[]> = {};
+    await Promise.all(
+      candidates.map(async (req) => {
+        try {
+          const record = await this.#dispatcher.retrieveCommit(req.cid);
+          if (this.#verifySignatures) {
+            await this.verifySignedCommit(record);
+          }
 
-    let req: Candidate = null
-    for (let index = 0; index < this._queue.length; index++) {
-      try {
-        req = this._queue[index]
-        const record = (await this._ceramic.ipfs.dag.get(req.cid)).value
-        if (this._verifySignatures) {
-          await this.verifySignedCommit(record)
-        }
-
-        const log = await this._loadCommitHistory(req.cid)
-        const candidate = new Candidate(new CID(req.cid), req.docId, log)
+          const log = await this._loadCommitHistory(req.cid);
+          const candidate = new Candidate(req.cid, req.docId, log);
 
-        if (!result[candidate.key]) {
-          result[candidate.key] = []
+          if (!result[candidate.key]) {
+            result[candidate.key] = [];
+          }
+          result[candidate.key].push(candidate);
+        } catch (e) {
+          console.error(e.message);
+          this._failCandidate(req, e.message);
         }
-        result[candidate.key].push(candidate)
-      } catch (e) {
-        console.error(e.message)
-        await this._failCandidate(req, e.message)
-      }
-    }
-
-    return result
+      })
+    );
+    return result;
   }
 
-  async _selectValidCandidates(groupedCandidates: Record<string, Candidate[]>): Promise<Candidate[]> {
```

Comment:
when I first set out to learn javascript a few months ago I read something that cautioned against mixing async and non-async functions in the same code.  I guess that isn't actually that big of a problem after all?  I don't recall what specifically the thing I read was worried could go wrong...